pyproject.toml

```diff
@@ -3,4 +3,5 @@ requires = [
     'scikit-build',
     'cython',
     'cmake',
+    'setuptools-scm',
```

Comment:
I'm not sure I totally understand what's going on in this change, but as I got it, adding this package changes something about how the build works. If that's the case, could a comment be added here that explains what we're relying on this for? I feel like otherwise it could be accidentally removed without understanding the context of this change.